src/CSET/operators/ageofair.py

```diff
@@ -0,0 +1,378 @@
+        least 2 time points to compute back trajectory.
+    YWIND: iris.cube.Cube
+        An iris cube containing the y component of wind on pressure levels, on a 0p5 degree grid.
+        Requires 4 dimensions, ordered time, pressure, latitude and longitude. Must contain at
+        least 2 time points to compute back trajectory.
+    WWIND: iris.cube.Cube
+        An iris cube containing the w component of wind on pressure levels, on a 0p5 degree grid.
+        Requires 4 dimensions, ordered time, pressure, latitude and longitude. Must contain at
+        least 2 time points to compute back trajectory.
+    GEOPOT: iris.cube.Cube
+        An iris cube containing geopotential height on pressure levels, on a 0p5 degree grid.
+        Requires 4 dimensions, ordered time, pressure, latitude and longitude. Must contain at
+        least 2 time points to compute back trajectory.
+    plev: int
+        The pressure level of which to compute the back trajectory on. The function will search to
+        see if this exists and if not, will raise an exception.
+    incW: bool
+        If incW is True, then the back trajectories will take into account vertical velocity fields
+        and thus a back trajectory takes a 3D path back. In this case, vertical wind fields are
+        smoothed in addition to the prior coarsening, to prevent instantaneous sampling of strong
+        updrafts/downdrafts that would advect a parcel to the bottom/top of atmosphere if the
+        temporal sampling of the data is coarse enough (e.g. hourly or 3 hourly). This option enables
+        the trajectory to capture large areas of rising/sinking air, in phenomena such as overturning
+        circulations. If incW if False, then compute back trajectories on a pressure level with no
+        vertical advection (essentially a laminar, stratified flow).
+    cyclic: book
+        If cyclic is True, then the code will assume no east/west boundary and if a back trajectory
+        reaches the boundary, it will emerge out of the other side. This option is useful for large
+        domains such as the K-SCALE tropical channel, where there are only north/south boundaries in
+        the domain.
+
+    Returns
+    -------
+    iris.cube.Cube
+        An iris cube of the age of air data, with 3 dimensions (time, latitude, longitude).
+
+    """
+    # If not None, then use multiple cores to run age of air diagnostic.
+    multicore = None
+
+    # Set up temporary directory to store intermediate age of air slices.
+    tmpdir = tempfile.mkdtemp()
+    logging.info("Made tmpdir", tmpdir)
+
+    # Check that all cubes are of same size (will catch different dimension orders too).
+    if not XWIND.shape == YWIND.shape == WWIND.shape == GEOPOT.shape:
+        raise ValueError("Cubes are not the same shape")
+
+    # Get time units and assign for later
+    if str(XWIND.coord("time").units) == "hours since 1970-01-01 00:00:00":
```

Comment:
```suggestion
    if str(XWIND.coord("time").units).startswith("hours since "):
```
This will handle more cases, as some models use the initiation time as the reference.

I feel like something clever should be possible with [cf-units.Unit](https://cf-units.readthedocs.io/en/latest/unit.html) (which is what Cube.units gives you), but the best I can think of is doing num2date of the first two points, and then doing `t1 - t0` to get the interval.